composite/crypto-volatility-index/src/cryptoVolatilityIndex.ts

```diff
@@ -0,0 +1,118 @@
+import { SigmaCalculator } from './sigmaCalculator'
+import { Big } from 'big.js'
+import moment from 'moment'
+const cryptoCurrencies = ['BTC', 'ETH']
+
+export const calculate = async (
+  oracleAddress: string,
+  multiply: number,
+  heartbeatMinutes: number,
+  isAdaptive: boolean,
+): Promise<number> => {
+  // Get all of the required derivatives data for the calculations, for all the relevant currencies
+  const derivativesData = await getDerivativesData(cryptoCurrencies)
+  // Calculate vix values for all currencies
+  const volatilityIndexData = await calculateVixValues(derivativesData)
+  // Apply weights to calculate the Crypto Vix
+  const weightedCVI = await calculateWeighted(volatilityIndexData)
+  // Smooth CVI with previous on-chain value if exists
+  const cvi = !isAdaptive
+    ? toOnChainValue(weightedCVI, multiply)
+    : await applySmoothing(weightedCVI, oracleAddress, multiply, heartbeatMinutes)
+
+  logger.info(`CVI: ${cvi}`)
+  validateIndex(cvi, multiply)
+  return cvi
+}
+
+const calculateVixValues = async (derivativesData: Record<string, CurrencyDerivativesData>) => {
+  const now = moment().utc()
+  const sigmaCalculator = new SigmaCalculator()
+  const vixValues = cryptoCurrencies.map((currency) => {
+    sigmaCalculator.sortByStrikePrice(derivativesData[currency])
+    const { e1, e2, exchangeRate, callsE1, putsE1, callsE2, putsE2 } = derivativesData[currency]
+    const weightedSigma: Big = sigmaCalculator.weightedSigma({
+      e1,
+      e2,
+      sigma1: sigmaCalculator.oneSigma(e1, exchangeRate, callsE1, putsE1, now),
+      sigma2: sigmaCalculator.oneSigma(e2, exchangeRate, callsE2, putsE2, now),
+      now,
+    })
+    return weightedSigma.sqrt().times(100)
+  })
+
+  return vixValues
+}
+
+const calculateWeighted = async (vixData: Array<Big>) => {
+  const dominanceByCurrency = await getDominanceByCurrency()
+  const weightedVix = cryptoCurrencies.reduce((vix, currency, idx) => {
+    const dominance = dominanceByCurrency[currency]
+    if (!dominance) throw new Error(`No dominance found for currency ${currency}`)
+    const currencyVix = new Big(vixData[idx])
+    // Weight by dominance
+    vix = vix.plus(currencyVix.times(new Big(dominance)))
+    return vix
+  }, new Big(0))
+
+  const weighted = Number(weightedVix.toFixed())
+  logger.debug(`Weighted volatility index:${weighted}`)
+  return weighted
+}
+
+const getDominanceByCurrency = async () => {
+  const dominanceProvider = util.getRequiredEnv('DOMINANCE_PROVIDER')
+  const dominanceAdapter = await getDominanceAdapter(dominanceProvider)
+  const dominanceData = await dominanceAdapter.getDominance(cryptoCurrencies)
+  return dominanceByCurrency(dominanceData)
+}
+
+const applySmoothing = async (
+  weightedCVI: number,
+  oracleAddress: string,
+  multiply: number,
+  heartBeatMinutes: number,
+): Promise<number> => {
+  const roundData = await getRpcLatestRound(oracleAddress, multiply)
```

Comment:
```suggestion
  const roundData = await getRpcLatestRound(oracleAddress)
```

We also should be dividing the `latestIndex` below by the multiply amount.

Reply:
done